### PR TITLE
[2.x] Refactor from query builder macros to laravel database expressions

### PR DESCRIPTION
**Pull request by ahawlitschek**
This pull request includes significant changes to the `laravel-magellan` package, focusing on simplifying the query building process, updating documentation, and removing deprecated methods. The most important changes include the removal of the `st`-prefixed builder functions, updates to the README to reflect these changes, and the addition of new utility methods and classes.

In our first version, we decided to add custom query builder methods via macro prefixed with `st` to gain access to the query builder while building the queries. This approach enabled us to use the default query binding feature. However, this method introduced complexity and i tight coupling of our postgis stuff to the query builder itself. 
When @saibotk switched to casters instead of the postgis trait, he also removed the automatic transformation of SRIDs. 
Having to do this manually now showed us, that our tight coupling was problematic, because we couldn't do stuff like this: 

```php
Port::create([
'location' => ST::transform($point, 4326)
]);
```

With our first approach, we need to do a separate query against the database, parse the result and so on. 
Quite a lot for something that should be seamless and easy. Also the readability of the code was quite a mess. 
With the Grammar in the `getValue(Grammar)`function in Laravel 10, another way appeared. 
https://github.com/tpetry/laravel-query-expressions explored this way and inspired us, to also change our approach.

Removing the tight coupling to the Builder also introduced some issues. 
1. Providing an Alias for the select was not that easy anymore. => Introduce the Aliased Expressen
2. Auto Cast was not possible without stSelect => add a new builder macro, that can handle casts, but need to be called explicitly

Dealing with the two issues, we think, that the newer syntax is much cleaner and easier to understand.

# TLDR;

### Simplification of Query Building:

* Removed `st`-prefixed builder functions such as `stSelect`, `stWhere`, `stOrderBy`, etc., from `BuilderMacros`. Now, default Laravel query builder methods should be used directly.

### Documentation Updates:

* Updated `README.md` to reflect the removal of `st`-prefixed methods and to provide examples using the default query builder methods along with the new `Aliased` expression class.

### New Additions:

* Added `Aliased` Expression class for wrapping expressions with `AS` in query selects.
* Introduced `withMagellanCasts()` method as an EloquentBuilder macro to automatically add the necessary casts for geometry types.

### Dependency Updates:

* Updated `composer.json` to remove the dependency on `spatie/invade`.

### Codebase Cleanup:

* Removed the entire `BuilderUtils` class as it is no longer necessary with the new query building approach.

### CODE REVIEW

File: src/Database/MagellanExpressions/MagellanBaseExpression.php [hunk 1]
```diff
@@ -64,43 +84,93 @@ public function invoke($builder, string $bindingType, ?string $as = null): Expre
...
+                    return $this->createSub($value).$geometryTypeCastAppend;
+                } elseif (is_array($value)) {
+                    $wrapped = array_map(fn ($geometry) => GeoParam::wrap($geometry), $value);
+                    $prepared = $this->prepareParams($grammar, $generator, $geometryTypeCastAppend, $wrapped);
+                    $imploded = implode(', ', $prepared);
+
+                    return "ARRAY[$imploded]";
+
+                } elseif ($value instanceof Geometry) {
+                    return $generator->toPostgisGeometrySql($value, Config::get('magellan.schema'));
```
Comment (saibotk) [inline src/Database/MagellanExpressions/MagellanBaseExpression.php:131]: We need to append the type cast for all other cases too.

**Review by saibotk — CHANGES_REQUESTED** [review]

File: tests/Models/SpatialQueriesTest.php [hunk 2]
```diff
@@ -201,9 +213,161 @@
...
+        ->st_buffer;
+
+    $parser = App::make(WKBParser::class);
+    $bufferedGeometry = $parser->parse($bufferedRawGeometry);
+
+    expect($bufferedGeometry)->toBeInstanceOf(Polygon::class);
+})->skip(message: 'Currently we only support closures and no Builder or Relations in parameters');
+
+// TODO: Add test case that uses relation and eloquent query builder instead of closure or query builder
+
```
[the remark this answers is not in the document]
Comment (ahawlitschek) [reply tests/Models/SpatialQueriesTest.php:287]: I added a test for the base database builder and skipped it with a reason in the description. 
Should we remove this also or should we also add skipped tests for the eloquent builder and relations?

File: src/Database/MagellanExpressions/MagellanBaseExpression.php (hunk 1 above)
Comment (ahawlitschek) [reply src/Database/MagellanExpressions/MagellanBaseExpression.php:131]: As discussed yesterday we might add a new way of casting. 
Maybe we will not need the type cast appends here in generell.
Let's continue our discussion later

**Review by saibotk — APPROVED** [review]
